packages/expo-updates/ios/Tests/EXUpdatesBuildDataTests.m

```diff
@@ -0,0 +1,192 @@
+    [_db openDatabaseInDirectory:_testDatabaseDir withError:&dbOpenError];
+    XCTAssertNil(dbOpenError);
+  });
+  
+  _manifest = [[EXManifestsNewManifest alloc] initWithRawManifestJSON:@{
+    @"runtimeVersion": @"1",
+    @"id": @"0eef8214-4833-4089-9dff-b4138a14f196",
+    @"createdAt": @"2020-11-11T00:17:54.797Z",
+    @"launchAsset": @{@"url": @"https://url.to/bundle.js", @"contentType": @"application/javascript"}
+  }];
+  
+  
+  _configChannelTestDictionary = @{
+    @"EXUpdatesScopeKey": scopeKey,
+    @"EXUpdatesURL": @"https://exp.host/@test/test",
+    @"EXUpdatesReleaseChannel": @"default",
+    @"EXUpdatesRequestHeaders": @{@"expo-channel-name":@"test"}
+  };
+  _configChannelTest = [EXUpdatesConfig configWithDictionary:_configChannelTestDictionary];
+  _configChannelTestTwoDictionary = @{
+    @"EXUpdatesScopeKey": scopeKey,
+    @"EXUpdatesURL": @"https://exp.host/@test/test",
+    @"EXUpdatesReleaseChannel": @"default",
+    @"EXUpdatesRequestHeaders": @{@"expo-channel-name":@"testTwo"}
+  };
+  _configChannelTestTwo = [EXUpdatesConfig configWithDictionary:_configChannelTestTwoDictionary
+  ];
+}
+
+- (void)tearDown
+{
+  dispatch_sync(_db.databaseQueue, ^{
+    [_db closeDatabase];
+  });
+  NSError *error;
+  [NSFileManager.defaultManager removeItemAtPath:_testDatabaseDir.path error:&error];
+  XCTAssertNil(error);
+}
+
+- (void)test_clearAllUpdatesFromDatabase {
+  dispatch_sync(_db.databaseQueue, ^{
+    EXUpdatesUpdate *update = [EXUpdatesNewUpdate updateWithNewManifest:_manifest response:nil config:_configChannelTest database:_db];
+  
+    NSError *updatesError;
+    [_db addUpdate:update error:&updatesError];
+    if (updatesError) {
+      XCTFail(@"%@", updatesError.localizedDescription);
+      return;
+    }
+
+    NSError *queryError;
+    NSArray<EXUpdatesUpdate *> *allUpdates = [_db allUpdatesWithConfig:_configChannelTest error:&queryError];
+    if (queryError) {
+      XCTFail(@"%@", queryError.localizedDescription);
+      return;
+    }
+    
+    XCTAssertGreaterThan(allUpdates.count, 0);
+  });
+  
+  [EXUpdatesBuildData clearAllUpdatesFromDatabase:_db config:_configChannelTest error:nil];
+  
+  dispatch_sync(_db.databaseQueue, ^{
+    NSError *queryError;
+    NSArray<EXUpdatesUpdate *> *allUpdatesAfter = [_db allUpdatesWithConfig:_configChannelTest error:&queryError];
+    if (queryError) {
+      XCTFail(@"%@", queryError.localizedDescription);
+      return;
+    }
+    
+    XCTAssertEqual(allUpdatesAfter.count, 0);
+  });
+}
+
+- (void)test_ensureBuildDataIsConsistent_buildDataIsNull {
+  // check no updates and build data is set
+
+  dispatch_sync(_db.databaseQueue, ^{
+    NSError *error;
+    NSDictionary *staticBuildData = [EXUpdatesBuildData getBuildDataFromDatabase:_db scopeKey:scopeKey error:&error];
+    XCTAssertNil(staticBuildData);
+
+    NSArray<EXUpdatesUpdate *> *allUpdates = [_db allUpdatesWithConfig:_configChannelTest error:&error];
+    XCTAssertEqual(allUpdates.count, 0);
```

Comment:
do you think it's worth testing the case where there are updates (ensuring they don't get cleared)?

Reply:
yea good point, I ended up adding an update to each test and checking that it was either cleared or left alone as intended.